### ISSUE TITLE
为什么识别模型做预测的时候，预测图片的数量数量还会影响预测的精度

### ISSUE THREAD

**Issue by dychb** [PaddlePaddle/PaddleOCR#6116]
![image](https://user-images.githubusercontent.com/59058681/166155070-e8780d65-45b9-46f2-9b57-97560e8bf659.png)
![image](https://user-images.githubusercontent.com/59058681/166155097-ad16d656-1178-4983-97b0-4f70edd1e29e.png)
同样的图片，我把后三张删了，前面三张图片的识别结果和精度就变了，有什么办法能解决吗，后者的结果明显要好

**Comment by Sanster** [comment]
推理是按 batch 推理的，不同宽度的图片会 padding 到同一宽度，两次跑的最大宽度不一样，这可能是结果不一样的原因，可以试试 batch_size 改成 1 以后结果是否一致

**Comment by dychb** [comment]
我在识别模型的配置文件里好像没有找到预测的batch_size，eval的batch_size调成1试了一下，对结果没什么影响，不知道在哪调预测的batch_size，是不是需要在代码里调
rec_icdar15_train.yml配置文件如下：
Global:
  use_gpu: true
  epoch_num: 72
  log_smooth_window: 20
  print_batch_step: 50
  save_model_dir: ./output/rec/ic15/
  save_epoch_step: 10
  # evaluation is run every 2000 iterations
  eval_batch_step: [0, 2000]
  cal_metric_during_train: True
  pretrained_model:
  checkpoints: #output/rec/ic15/latest
  save_inference_dir: ./
  use_visualdl: False
  infer_img: doc/imgs_words_en/word_10.png
  # for data or label process
  character_dict_path: ppocr/utils/ic15_dict.txt
  max_text_length: 11
  infer_mode: False
  use_space_char: False
  save_res_path: ./output/rec/predicts_ic15.txt

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.005
    name: Cosine
  regularizer:
    name: 'L2'
    factor: 0

Architecture:
  model_type: rec
  algorithm: CRNN
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 96
  Head:
    name: CTCHead
    fc_decay: 0

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: train_data/train_rec_340k
    label_file_list: ["train_data/train_rec_label_340k.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 100]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 128
    drop_last: True
    num_workers: 8
    use_shared_memory: False

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: train_data/train_rec_340k
    label_file_list: ["train_data/train_rec_label_340k.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 100]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1
    num_workers: 4
    use_shared_memory: False

**Comment by tink2123** [comment]
在推理的时候设置识别bs=1，此时不需要修改配置文件了，命令如下：
```
python3 tools/infer/predict_rec.py --image_dir="./doc/imgs_words/ch/word_4.jpg" --rec_model_dir="./ch_PP-OCRv3_rec_infer/" --rec_batch_num=1
```
